### DISCUSSION TITLE
Any ideas to convert the inserted datasets into BOOKSHELF format

### DISCUSSION THREAD

**Discussion by xiaosiCU** [The-OpenROAD-Project/OpenROAD#3708] [General]
I'm focusing on macro placement task and familiar to bookshelf format data processing.
Bookshelf format provides convenience for extracting macros and netlist information.
I do know there are scripts converting .lef.def files into bookshelf format, but how about the inserted datasets in the OpenROAD proj? Like ariane133 using nangate45 platform.

**Comment by maliberty** [comment]
You would have to convert to lef/def.  I don't know of a tool to do so automatically.  Alternatively you could parse and build the db in TCL if that's easier.
